numpy/core/fromnumeric.py

```diff
@@ -1404,22 +1411,21 @@ def resize(a, new_shape):
     """
     if isinstance(new_shape, (int, nt.integer)):
         new_shape = (new_shape,)
+
     a = ravel(a)
-    Na = len(a)
-    total_size = um.multiply.reduce(new_shape)
-    if Na == 0 or total_size == 0:
-        return mu.zeros(new_shape, a.dtype)
 
-    n_copies = int(total_size / Na)
-    extra = total_size % Na
+    new_size = 1
+    for dim_length in new_shape:
+        new_size *= dim_length
```

Comment:
Could put this below the error check for a marginal speed boost in the error case, but doesn't really matter.

Reply:
Doubt its nicer to read, and I refuse to optimize error-cases without good reason :). Thanks for another pass Eric lets put it in.